### PR TITLE
fix: allow user gesture restricted code to be run in page.evaluate

### LINKED ISSUE #2502: Page.evaluate should bypass user gestures
`await page.evaluate('document.documentElement.webkitRequestFullscreen()')` currently does not work because full screen requires a user gesture. Running the same snippet in the DevTools console does work. 

Presumably devtools bypasses user gestures by adding [`userGesture: true`](https://chromedevtools.github.io/devtools-protocol/tot/Runtime#method-evaluate). We can do the same for our `Runtime.evaluate` and `Runtime.callFunctionOn` calls.

### PR DESCRIPTION
**Pull request by JoelEinbinder**
Fixes #2502

### CODE REVIEW

File: test/page.spec.js [hunk 1]
```diff
@@ -196,6 +196,13 @@ module.exports.addTests = function({testRunner, expect, puppeteer, DeviceDescrip
       const isFive = await page.evaluate(e => Object.is(e, 5), aHandle);
       expect(isFive).toBeTruthy();
     });
+    it('should simulate a user gesture', async({page, server}) => {
+      await page.evaluate(() => {
+        const audio = document.createElement('audio');
+        audio.src = 'data:audio/wav;base64,UklGRiQAAABXQVZFZm10IBAAAAABAAEARKwAAIhYAQACABAAZGF0YQAAAAA=';
+        return audio.play();
```
Comment (JoelEinbinder) [inline test/page.spec.js]: This returns a promise, which throws if the audio fails to play.

File: lib/ExecutionContext.js [hunk 2]
```diff
@@ -75,7 +75,8 @@ class ExecutionContext {
       executionContextId: this._contextId,
       arguments: args.map(convertArgument.bind(this)),
       returnByValue: false,
-      awaitPromise: true
+      awaitPromise: true,
+      userGesture: true
```
Comment (ebidel) [inline lib/ExecutionContext.js:85]: Do we need this up on 67 too for the `Runtime.evaluate` call?

File: test/page.spec.js (hunk 1 above)
Comment (ebidel) [reply test/page.spec.js]: I'd add that comment. It's not obvious unless you know about our autoplay policy.

File: test/page.spec.js [hunk 3]
```diff
@@ -196,6 +196,13 @@ module.exports.addTests = function({testRunner, expect, puppeteer, DeviceDescrip
       const isFive = await page.evaluate(e => Object.is(e, 5), aHandle);
       expect(isFive).toBeTruthy();
     });
+    it('should simulate a user gesture', async({page, server}) => {
+      await page.evaluate(() => {
```
Comment (ebidel) [inline test/page.spec.js]: add a test of for the string case: page.evaluate('...')

File: lib/ExecutionContext.js (hunk 2 above)
Comment (JoelEinbinder) [reply lib/ExecutionContext.js:85]: Good catch. done.

File: test/page.spec.js (hunk 3 above)
Comment (JoelEinbinder) [reply test/page.spec.js]: done.

File: test/page.spec.js (hunk 1 above)
Comment (JoelEinbinder) [reply test/page.spec.js]: done

**Review by ebidel — APPROVED** [review]

**Pull request merged**